OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -56,9 +58,19 @@ public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttribute
             Attributes = userAttributes ?? new UserAttributes();
             ForcedDecisionsStore = forcedDecisionsStore ?? new ForcedDecisionsStore();
             UserId = userId;
+            QualifiedSegments = new List<string>();
         }
 
         private OptimizelyUserContext Copy() => new OptimizelyUserContext(Optimizely, UserId, GetAttributes(), GetForcedDecisionsStore(), ErrorHandler, Logger);
+        
+        /// <summary>
+        /// Returns true if the user is qualified for the given segment name
+        /// </summary>
+        /// <param name="segment">A String segment key which will be check in qualified segments list that if it exist then user is qualified.</param>
+        /// <returns>Is user qualified for a segment.</returns>
+        public bool IsQualifiedFor(string segment) {
+            return QualifiedSegments.Contains(segment);
```

Comment:
`QualifiedSegments` is `private` and `readonly` at class instantiation.  This should make it thread safe since it cannot be modified and used only within the class.

Reply:
I don't think we need to manually `lock()` or move to a thread-safe collection do we @msohailhussain ?